core/new-gui/src/app/workspace/component/result-panel/visualization-frame/visualization-frame.component.ts

```diff
@@ -12,27 +15,49 @@ import { VisualizationFrameContentComponent } from "../../visualization-panel-co
   selector: "texera-visualization-frame",
   templateUrl: "./visualization-frame.component.html",
   styleUrls: ["./visualization-frame.component.scss"],
+  providers: [{ provide: NZ_ICONS, useValue: [FullscreenExitOutline, FullscreenOutline] }],
 })
 export class VisualizationFrameComponent {
   @Input() operatorId?: string;
+  @ViewChild("modalTitle") modalTitle: TemplateRef<{}>;
   modalRef?: NzModalRef;
+  isFullscreen: Boolean = false;
 
-  constructor(private modalService: NzModalService) {}
+  constructor(private modalService: NzModalService) {
+    this.modalTitle = {} as TemplateRef<any>;
+  }
 
   onClickVisualize(): void {
     if (!this.operatorId) {
       return;
     }
-
     this.modalRef = this.modalService.create({
-      nzTitle: "Visualization",
-      nzStyle: { top: "20px" },
-      nzWidth: 1100,
-      nzFooter: null, // null indicates that the footer of the window would be hidden
+      nzTitle: this.modalTitle,
+      nzStyle: { top: "20px", width: "70vw", height: "78vh" },
       nzContent: VisualizationFrameContentComponent,
+      nzFooter: null, // null indicates that the footer of the window would be hidden
+      nzBodyStyle: { width: "70vw", height: "74vh" },
       nzComponentParams: {
         operatorId: this.operatorId,
       },
     });
   }
+
+  toggleFullscreen(): void {
+    this.isFullscreen = !this.isFullscreen;
+    if (!this.modalRef) {
```

Comment:
It might not be a good idea to use `!this.modalRef`. Please be careful with typescript/javascript falsy value evaluation. read more [here](https://stackoverflow.com/questions/5113374/javascript-check-if-variable-exists-is-defined-initialized).